docs/how-does-version-resolution-work-in-maven-and-gradle.md

```diff
@@ -0,0 +1,108 @@
+# How does version resolution work in Maven and Gradle?
+
+The build system that you choose for building your Java code determines which
+version selection algorithm is used to choose the versions of your
+dependencies. Unfortunately, the two most popular build systems in open source
+(Maven and Gradle) use different version resolution algorithms. No matter which
+of the two you choose, you are likely to have direct or indirect dependencies
+that use the other one, so you need to understand how they both work when your
+dependency graph experiences conflicts.
+
+Before we look at Maven and Gradle specifically, we need to understand what
+happens in the version resolution process. The input of this process is a
+dependency graph which reflects the dependencies exactly as specified by each
+library in the graph, which we call an unresolved dependency graph. In such a
+graph, there can be multiple versions of each library. The version resolution
+process walks the unresolved dependency graph and produces a new graph where
```

Comment:
I'm not sure it does, at least not Maven. It produces a linear classpath, not another graph.

Reply:
Ok, I will detach the actual output of Maven from the treatment of the output as a new graph.